From 6fecea21c157d55920610cba84cdbf5d837b2b4a Mon Sep 17 00:00:00 2001
From: Jiaxun Yang <jiaxun.yang@flygoat.com>
Date: Thu, 19 Jan 2023 20:42:56 +0000
Subject: [PATCH 140/140] AOSCOS: wifi: rt2x00: Condition interface type
 getters with config options

When compiling this driver for platforms witt partial clk
support (i.e. MIPS/Loongson64), it will report missing symbol
for clk_get_sys.

Conditioning interface type getters will help compiler to eliminate
unsed code and thus exclude calls to clk symbols. Also hopefully
reduce code size on other platforms.

Signed-off-by: Jiaxun Yang <jiaxun.yang@flygoat.com>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/net/wireless/ralink/rt2x00/rt2x00.h | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/drivers/net/wireless/ralink/rt2x00/rt2x00.h b/drivers/net/wireless/ralink/rt2x00/rt2x00.h
index a90a518b40d3..010f8e911f75 100644
--- a/drivers/net/wireless/ralink/rt2x00/rt2x00.h
+++ b/drivers/net/wireless/ralink/rt2x00/rt2x00.h
@@ -1164,23 +1164,27 @@ static inline bool rt2x00_intf(struct rt2x00_dev *rt2x00dev,
 
 static inline bool rt2x00_is_pci(struct rt2x00_dev *rt2x00dev)
 {
-	return rt2x00_intf(rt2x00dev, RT2X00_CHIP_INTF_PCI) ||
-	       rt2x00_intf(rt2x00dev, RT2X00_CHIP_INTF_PCIE);
+	return IS_ENABLED(CONFIG_RT2X00_LIB_PCI) &&
+		(rt2x00_intf(rt2x00dev, RT2X00_CHIP_INTF_PCI) ||
+		rt2x00_intf(rt2x00dev, RT2X00_CHIP_INTF_PCIE));
 }
 
 static inline bool rt2x00_is_pcie(struct rt2x00_dev *rt2x00dev)
 {
-	return rt2x00_intf(rt2x00dev, RT2X00_CHIP_INTF_PCIE);
+	return IS_ENABLED(CONFIG_RT2X00_LIB_PCI) &&
+		rt2x00_intf(rt2x00dev, RT2X00_CHIP_INTF_PCIE);
 }
 
 static inline bool rt2x00_is_usb(struct rt2x00_dev *rt2x00dev)
 {
-	return rt2x00_intf(rt2x00dev, RT2X00_CHIP_INTF_USB);
+	return IS_ENABLED(CONFIG_RT2X00_LIB_USB) &&
+		rt2x00_intf(rt2x00dev, RT2X00_CHIP_INTF_USB);
 }
 
 static inline bool rt2x00_is_soc(struct rt2x00_dev *rt2x00dev)
 {
-	return rt2x00_intf(rt2x00dev, RT2X00_CHIP_INTF_SOC);
+	return IS_ENABLED(CONFIG_RT2X00_LIB_SOC) &&
+		rt2x00_intf(rt2x00dev, RT2X00_CHIP_INTF_SOC);
 }
 
 /* Helpers for capability flags */
-- 
2.47.0.rc0

